### PR TITLE
search: Add randomize users on tag search.

### LINKED ISSUE #3380: [FEATURE] Randomise Users on tags search
### Description

The Profiles on the search display are sorted alphabetically based on the profile's GitHub username
This causes all the profiles starting with the letter `A` to come on top of others
It'd be much better if this section was randomized so that more profiles can be discovered

### Screenshots

`Prakhar Goyal`'s profile is on top here as it's being sorted using the GitHub username as `Prakhar Goyal`'s username is `0xPrakharG`, it comes on top

image

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by Joelute**
## Fixes Issue
Closes #3380 

## Changes proposed

Currently users are sorted alphabetically based on the user's Github username.
An randomizer is added to help randomize the users that show up and help 
promote more profiles to be discovered. 

## Check List (Check all the applicable boxes) 

- [ ] My code follows the code style of this project.
- [ ] My change requires changes to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] All new and existing tests passed.
- [x] This PR does not contain plagiarized content.
- [x] The title of my pull request is a short description of the requested changes.

## Screenshots

image
image

## Note to reviewers

<a href="https://gitpod.io/#https://github.com/EddieHubCommunity/LinkFree/pull/4521"></a>

### CODE REVIEW

**Review by priyankarpal — DISMISSED** [review]
Awesome 🎉 Thanks for your first contribution.

**Review by iamthecloverly — DISMISSED** [review]

**Comment by Joelute** [comment]
Pushed new commit to resolve conflicting changes

File: pages/search.js [hunk 1]
```diff
@@ -77,7 +87,7 @@ export default function Search({ data }) {
       setNotFound();
     }
 
-    setFilteredUsers(results);
+    setFilteredUsers(randomizeData(results));
```
Comment (eddiejaoude) [inline pages/search.js:90]: I think a simpler way might be to remove the custom function and use the `sort` function from javascript ...

```suggestion
    setFilteredUsers(results.sort(() => 0.5 - Math.random()));
```

**Review by eddiejaoude — CHANGES_REQUESTED** [review]
Suggested simplifying the code, see inline comments

File: pages/search.js (hunk 1 above)
Comment (Joelute) [reply pages/search.js:90]: Wow, that's a great solution. Will make the changes right away!

Comment (eddiejaoude) [reply pages/search.js]: But you implemented it differently to the example I shared? 

You can just accept the "suggestion" if you agree with the suggestion

**Review by eddiejaoude — APPROVED** [review]
Thanks 👍

**Pull request merged**